### PR TITLE
Feature/contact view

### PR DESCRIPTION
**Pull request by Seeru-crypto**

### CODE REVIEW

File: aurora/src/components/contact/ContactIcon.tsx [hunk 1]
```diff
@@ -1,5 +1,70 @@
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
 import { ContactIconInterface } from './Contact';
 
 export default function ({ icon }: { icon: ContactIconInterface }) {
-    return <div>{icon.name}</div>;
+    // const [windowObjectReference, setWindowObjectReference] =
+    //     useState<any>(null);
+    // const [previousUrl, setPreviousUrl] = useState<string>('');
```
Comment (Seeru-crypto) [inline aurora/src/components/contact/ContactIcon.tsx:9]: remove dead code

File: aurora/src/components/contact/Contact.tsx [hunk 2]
```diff
@@ -1,56 +1,68 @@
...
+    }, [isToastShown]);
+
+    const dispatch = useAppDispatch();
+
+    return (
+        <section id="contact" className={styles.container}>
+            <div className={styles.contactMain}>
+                <h1 className={styles.heading}>Get in touch with me!</h1>
+                <section className={styles.contactText}>
+                    You can share your impressions on why would someone need
```
Comment (Seeru-crypto) [inline aurora/src/components/contact/Contact.tsx:45]: Import the fluff text from data.json

File: aurora/src/components/contact/emailDisplayField.tsx [hunk 3]
```diff
@@ -1,13 +1,21 @@
 import React from 'react';
 import styled from 'styled-components';
```
Comment (Seeru-crypto) [inline aurora/src/components/contact/emailDisplayField.tsx:3]: Refacto the component into a more general component, which takes a text field as a prop

File: aurora/src/components/util/Button.tsx [hunk 4]
```diff
@@ -1,62 +1,61 @@
...
+        props.bordered &&
+        `
   background: blue;
   `}
 
-  ${(props) => props.bordered && BorderedStyle};
-  ${(props) => props.bordered && ToxicButtonStyle}
+    ${(props) => props.bordered && ToxicButtonStyle}
+    ${(props) => props.bordered && BorderedStyle};
 `;
```
Comment (Seeru-crypto) [inline aurora/src/components/util/Button.tsx:48]: Remove unnessecary styles

File: aurora/src/styles/Contact.module.css [hunk 5]
```diff
@@ -24,11 +24,17 @@
 .contactButtons {
     display: flex;
     flex-direction: row;
-    gap: 0.5rem;
+    gap: 1rem;
```
Comment (Seeru-crypto) [inline aurora/src/styles/Contact.module.css:27]: Move the css styling from .module file to a styled component within Contact.tsx

File: aurora/src/pages/index.tsx [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
-import { useEffect, useRef, useState } from 'react';
+import { useEffect, useRef } from 'react';
 import {
     About,
     Contact,
     Experience,
     LandingPage,
     Work,
 } from '../components/index';
+import Toast from '../components/util/Toast';
```
Comment (Seeru-crypto) [inline aurora/src/pages/index.tsx:10]: Create util index

**Comment by Seeru-crypto** [comment]
Merge resume buttons into a single component, which changes style depending on prop input

File: aurora/src/components/contact/Contact.tsx [hunk 7]
```diff
@@ -1,56 +1,113 @@
...
-          )}
-        </div>
-      </div>
-      {/* <div className={styles.contactIcons}>
+
+const Contact = React.forwardRef ((props, ref: ForwardedRef<HTMLElement>) : JSX.Element => {
+    const icons: ContactIconInterface[] = config.ICONS;
+    const resumeLinkData: ExternalLinkInterface = {
+        onClick: config.CV_DOWNLOAD_LINK,
+        label: 'get my stuff',
```
Comment (ristokitsing) [inline aurora/src/components/contact/Contact.tsx:21]: More like `git` your stuff

File: aurora/src/components/contact/ContactIcon.tsx [hunk 8]
```diff
@@ -1,5 +1,27 @@
+import Image from 'next/image';
+import styled from 'styled-components';
 import { ContactIconInterface } from './Contact';
 
-export default function ({ icon }: { icon: ContactIconInterface }) {
-    return <div>{icon.name}</div>;
+export default function ContactIcon ({ icon }: { icon: ContactIconInterface }) {
+    return (
+        <ImageStyle href={icon.href} target="_blank">
+            <Image
```
Comment (ristokitsing) [inline aurora/src/components/contact/ContactIcon.tsx:8]: As an `Image`, this is fine, yea, NextJS has their own super-optimized tag for that

File: aurora/src/components/contact/ContactIcon.tsx [hunk 9]
```diff
@@ -1,5 +1,27 @@
...
+    );
 }
+
+const ImageStyle = styled.a`
+    height: 25px;
+    width: 25px;
+    filter: grayscale();
+    cursor: pointer;
+
+    &&:hover {
```
Comment (ristokitsing) [inline aurora/src/components/contact/ContactIcon.tsx:24]: In `styled-components`, you don't actually need `&&`

File: aurora/src/pages/index.tsx [hunk 10]
```diff
@@ -29,21 +32,57 @@ export default function Home({
...
+
+        return () => {
+            // ToDo fix potential error, where ref is already changed by the time this function runs
+            if (landingRef.current) observer.unobserve(landingRef.current);
+            if (aboutRef.current) observer.unobserve(aboutRef.current);
+            if (experienceRef.current) observer.unobserve(experienceRef.current);
+            if (projectRef.current) observer.unobserve(projectRef.current);
+            if (contactRef.current) observer.unobserve(contactRef.current);
+        }
+    }, [aboutRef, experienceRef, projectRef, landingRef, contactRef, dispatch]);
```
Comment (ristokitsing) [inline aurora/src/pages/index.tsx:85]: I kinda don't like how for each and everyone you have to be so declarative but on top of my head I also don't have a better solution; needs thought

**Review by ristokitsing — APPROVED** [review]

**Pull request merged**